SpatialGDK/Documentation/content/singleton-actors.md

```diff
@@ -35,30 +30,21 @@ class TESTSUITE_API AExampleGameGameState : public AGameStateBase
 }
 ```
 
-### Streaming queries
-
-To make sure all server-workers check out Singleton Actor entities, you need to configure the worker to have [streaming queries](https://docs.improbable.io/reference/latest/shared/worker-configuration/bridge-config#streaming-queries) for each Singleton Actor’s components.
-
-In our example with `ExampleGameGameState`, the Schema Generator creates a schema component called `ExampleGameGameStateMultiClientRepData`. You need to add this as a streaming query to the worker configuration file (spatial/workers/unreal/spatialos.UnrealWorker.worker.json).
-
-In the `bridge` field of the worker configuration file, there should be a section that looks like this:
+The code snippet below shows how to tag a native C++ class with the appropriate Private Singleton identifiers.
 
 ```
-"streaming_query": [
-      {
-        "global_component_streaming_query": {
-          "component_name": "improbable.unreal.GlobalStateManager"
-        }
-      },
-      {
-        "global_component_streaming_query": {
-          "component_name": "improbable.unreal.generated.examplegamegamestate.ExampleGameGameStateMultiClientRepData"
-        }
-      }
-    ],
+UCLASS(SpatialType=(Singleton, ServerOnly))
+class TESTSUITE_API AExampleGameGameMode : public AGameModeBase
+{
+  GENERATED_BODY()
+  ...
+}
 ```
 
-This creates two streaming queries, one for the `GlobalStateManager` and one for the `ExampleGameGameState` component. For each Singleton Actor you register, you need to add another streaming query for that Singleton Actor’s `MultiClientRepData` component. We understand this workflow is a little clumsy and will be improved in the future.
+To tag a blueprint class as a Public Singleton, open the class in the blueprint editor and navigate to the `Class Settings`. In the `Advanced` section inside `Class Options`, tick the `Spatial Type` checkbox and add `Singleton` to the `Spatial Description` textbox. To tag a blueprint class as a Private Singleton, follow the same steps, but add `ServerOnly` to the `Spatial Description` textbox.
+
+This is an example of what your blueprint `Class Options` should look like if you've tagged it as a Private Singleton:
+![Singleton blueprint]({{assetRoot}}assets/screen-grabs/blueprint_singleton.png)
 
-And that's it! You have successfully specified a Singleton Actor. Make sure you generate a new snapshot and schema using the [SpatialOS GDK for Unreal toolbar]({{urlRoot}}/content/toolbar).
+And that's it! You have successfully specified a Singleton Actor. Make sure you generate schema and create a new snapshot using the [SpatialOS GDK for Unreal toolbar]({{urlRoot}}/content/toolbar).
```

Comment:
Suggest link to the glossary and using the slightly shorter name for the toolbar (have put in backticks to keep the link exposed. I am not suggesting this is a good look for this bit of text). :

`Make sure you generate [schema]({{urlRoot}}/content/glossary#schema) and create a new [snapshot]( ({{urlRoot}}/content/generating-a-snapshot) using the [SpatialOS GDK toolbar]({{urlRoot}}/content/toolbar).`